functions/New-DbaAgentJobStep.ps1

```diff
@@ -0,0 +1,341 @@
+                }
+
+                if ($OnSuccessAction) {
+                    Write-Message -Message "Setting job step success action to $OnSuccessAction" -Level Verbose 
+                    $JobStep.OnSuccessAction = $OnSuccessAction
+                }
+
+                if ($OnSuccessStepId) {
+                    Write-Message -Message "Setting job step success step id to $($OnSuccessStepId)" -Level Verbose 
+                    $JobStep.OnSuccessStep = $OnSuccessStepId
+                }
+
+                if ($OnFailAction) {
+                    Write-Message -Message "Setting job step fail action to $OnFailAction" -Level Verbose 
+                    $JobStep.OnFailAction = $OnFailAction
+                }
+
+                if ($OnFailStepId) {
+                    Write-Message -Message "Setting job step fail step id to $($OnFailStepId)" -Level Verbose 
+                    $JobStep.OnFailStep = $OnFailStepId
+                }
+
+                if ($DatabaseName) {
+                    # Check if the database is present on the server
+                    if (($Server.Databases).Name -contains $DatabaseName) {
+                        Write-Message -Message "Setting job step database name to $DatabaseName" -Level Verbose 
+                        $JobStep.DatabaseName = $DatabaseName
+                    }
+                    else {
+                        Stop-Function -Message ("The database is not present on instance $instance.") -Target $JobName
+                        return
+                    }
+                }
+
+                if ($DatabaseUserName -and $DatabaseName) {
+                    # Check if the username is present in the database
+                    if (($Server.Databases[$DatabaseName].Users).Name -contains $DatabaseUserName) {
+                    
+                        Write-Message -Message "Setting job step database username to $DatabaseUserName" -Level Verbose 
+                        $JobStep.DatabaseUserName = $DatabaseUserName
+                    }
+                    else {
+                        Stop-Function -Message ("The database user is not present in the database $DatabaseName on instance $instance.") -Target $StepName
+                        return
+                    }
+                }
+
+                if ($RetryAttempts) {
+                    Write-Message -Message "Setting job step retry attempts to $($RetryAttempts)" -Level Verbose 
+                    $JobStep.RetryAttempts = $RetryAttempts
+                }
+
+                if ($RetryInterval) {
+                    Write-Message -Message "Setting job step retry interval to $($RetryInterval)" -Level Verbose 
+                    $JobStep.RetryInterval = $RetryInterval
+                }
+
+                if ($OutputFileName) {
+                    Write-Message -Message "Setting job step output file name to $OutputFileName" -Level Verbose 
+                    $JobStep.OutputFileName = $OutputFileName
```

Comment:
Ok, found it. We need to add the ability to set the [JobStepFlags property](https://msdn.microsoft.com/en-us/library/microsoft.sqlserver.management.smo.agent.jobstepflags.aspx).

I think I would just do a specific parameter as `-JobStepFlags` because you can have multiple options (e.g. log to a file and a table). Use validate set so only these can be selected and then the help we may just need to add a more descriptive explanation of them?

| Member name                        | Description                                                                 |
|------------------------------------|-----------------------------------------------------------------------------|
| AppendAllCmdExecOutputToJobHistory | Job history, including command output, is appended to the job history file. |
| AppendToJobHistory                 | Job history is appended to the job history file.                            |
| AppendToLogFile                    | Job history is appended to the SQL Server log file.                         |
| AppendToTableLog                   | Job history is appended to a log table.                                     |
| LogToTableWithOverwrite            | Job history is written to a log table, overwriting previous contents.       |
| None                               | Job history is not appended to a file.                                      |
| ProvideStopProcessEvent            | Job processing is stopped.                                                  |